datafusion/proto/src/from_proto.rs

```diff
@@ -218,20 +218,25 @@ impl From<protobuf::PrimitiveScalarType> for DataType {
             protobuf::PrimitiveScalarType::Float64 => DataType::Float64,
             protobuf::PrimitiveScalarType::Utf8 => DataType::Utf8,
             protobuf::PrimitiveScalarType::LargeUtf8 => DataType::LargeUtf8,
+            protobuf::PrimitiveScalarType::Binary => DataType::Binary,
+            protobuf::PrimitiveScalarType::LargeBinary => DataType::LargeBinary,
             protobuf::PrimitiveScalarType::Date32 => DataType::Date32,
-            protobuf::PrimitiveScalarType::TimeMicrosecond => {
-                DataType::Time64(TimeUnit::Microsecond)
-            }
-            protobuf::PrimitiveScalarType::TimeNanosecond => {
+            protobuf::PrimitiveScalarType::Time64 => {
                 DataType::Time64(TimeUnit::Nanosecond)
             }
+            protobuf::PrimitiveScalarType::TimestampMicrosecond => {
```

Comment:
I changed the names of `PrimitiveScalarType` from `Time` here to `Timestamp` be consistent with the `ScalarValue` variants as well as the arrow type system